core/domain/prod_validation_jobs_one_off.py

```diff
@@ -110,67 +351,3183 @@ def validate(cls, item):
+            if item.exploration_model_last_updated != last_human_update_time:
+                cls.errors['exploration model last updated check'].append((
+                    'Entity id %s: The exploration_model_last_updated '
+                    'field: %s does not match the last time a commit was '
+                    'made by a human contributor: %s') % (
+                        item.id, item.exploration_model_last_updated,
+                        last_human_update_time))
+
+    @classmethod
+    def _validate_related_model_properties(cls, item):
+        """Validate that model properties match the corresponding exploration
+        model and exploration rights model properties.
+
+        Args:
+            item: ExpSummaryModel to validate.
+        """
+        _, exploration_model_tuples_list = cls.external_models[
+            'exploration_ids']
+        _, exploration_rights_model_tuples_list = cls.external_models[
+            'exploration_rights_ids']
+
+        exploration_model_properties_dict = {
+            'title': 'title',
+            'category': 'category',
+            'objective': 'objective',
+            'language_code': 'language_code',
+            'tags': 'tags',
+            'exploration_model_created_on': 'created_on',
+        }
+
+        for (_, exploration_model) in exploration_model_tuples_list:
+            for (property_name, exploration_model_property_name) in (
+                    exploration_model_properties_dict.iteritems()):
+                value_in_summary_model = getattr(item, property_name)
+                value_in_exploration_model = getattr(
+                    exploration_model, exploration_model_property_name)
+
+                summary_model_output_value = value_in_summary_model
+                if isinstance(summary_model_output_value, list):
+                    summary_model_output_value = (',').join(
+                        summary_model_output_value)
+
+                exploration_model_output_value = (
+                    value_in_exploration_model)
+                if isinstance(exploration_model_output_value, list):
+                    exploration_model_output_value = (',').join(
+                        exploration_model_output_value)
+
+                if value_in_summary_model != value_in_exploration_model:
+                    cls.errors['%s field check' % property_name].append((
+                        'Entity id %s: %s field in entity: %s does not match '
+                        'corresponding exploration %s field: %s') % (
+                            item.id, property_name, summary_model_output_value,
+                            exploration_model_property_name,
+                            exploration_model_output_value))
+
+
+        exploration_rights_model_properties_dict = {
+            'first_published_msec': 'first_published_msec',
+            'status': 'status',
+            'community_owned': 'community_owned',
+        
```

Comment:
This is a little confusing. Why is there so much logic needed to figure out which part is the exploration_id? I think the logic ought to be simpler...

Ditto below.

Reply:
There was a mistake earlier. I have tried making the logic simpler and readable now. Does that look ok?

Reply:
I think what I'm wondering is is, why do you need "if"/"else" blocks here? Shouldn't we be expecting all file model IDs to have a consistent format that can be introspected directly?

/cc @aks681

Reply:
Hi @seanlip, some changes were not pushed and there are some updates to finding the id. Can you PTAL now, since the if/else is removed. Thanks!